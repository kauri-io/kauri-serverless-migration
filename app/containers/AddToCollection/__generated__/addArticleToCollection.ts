/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { ResourceIdentifierInput } from "./../../../../__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: addArticleToCollection
// ====================================================

export interface addArticleToCollection_addCollectionResource {
  __typename: "MutationResponse";
  hash: string | null;
}

export interface addArticleToCollection {
  addCollectionResource: addArticleToCollection_addCollectionResource | null;
}

export interface addArticleToCollectionVariables {
  id?: string | null;
  sectionId?: string | null;
  resourceId?: ResourceIdentifierInput | null;
  position?: number | null;
}
