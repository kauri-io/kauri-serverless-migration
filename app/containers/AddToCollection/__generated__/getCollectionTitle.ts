/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: getCollectionTitle
// ====================================================

export interface getCollectionTitle_getCollection {
  __typename: "CollectionDTO";
  name: string | null;
}

export interface getCollectionTitle {
  getCollection: getCollectionTitle_getCollection | null;
}

export interface getCollectionTitleVariables {
  id?: string | null;
}
