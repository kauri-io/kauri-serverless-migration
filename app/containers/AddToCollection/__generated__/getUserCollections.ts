/* tslint:disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: getUserCollections
// ====================================================

export interface getUserCollections_searchCollections_content_sections_resources_CommunityDTO {
  __typename:
    | "CommunityDTO"
    | "PublicUserDTO"
    | "CommentDTO"
    | "CommunityMemberDTO"
    | "TemplateDTO"
    | "CuratedListDTO"
    | "CollectionDTO"
    | "SearchResultDTO"
    | "UserDTO";
}

export interface getUserCollections_searchCollections_content_sections_resources_ArticleDTO {
  __typename: "ArticleDTO";
  id: string | null;
  version: number | null;
}

export type getUserCollections_searchCollections_content_sections_resources =
  | getUserCollections_searchCollections_content_sections_resources_CommunityDTO
  | getUserCollections_searchCollections_content_sections_resources_ArticleDTO;

export interface getUserCollections_searchCollections_content_sections {
  __typename: "SectionDTO";
  id: string | null;
  name: string | null;
  resources:
    | (getUserCollections_searchCollections_content_sections_resources | null)[]
    | null;
}

export interface getUserCollections_searchCollections_content {
  __typename: "CollectionDTO";
  id: string | null;
  name: string | null;
  sections:
    | (getUserCollections_searchCollections_content_sections | null)[]
    | null;
}

export interface getUserCollections_searchCollections {
  __typename: "ResponsePage_CollectionDTO";
  content: (getUserCollections_searchCollections_content | null)[] | null;
}

export interface getUserCollections {
  searchCollections: getUserCollections_searchCollections | null;
}

export interface getUserCollectionsVariables {
  userId?: string | null;
}
