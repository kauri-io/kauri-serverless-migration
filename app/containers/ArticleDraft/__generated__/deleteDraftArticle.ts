/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: deleteDraftArticle
// ====================================================

export interface deleteDraftArticle_cancelArticle {
  __typename: "MutationResponse";
  hash: string | null;
}

export interface deleteDraftArticle {
  cancelArticle: deleteDraftArticle_cancelArticle | null;
}

export interface deleteDraftArticleVariables {
  id?: string | null;
  version?: number | null;
}
