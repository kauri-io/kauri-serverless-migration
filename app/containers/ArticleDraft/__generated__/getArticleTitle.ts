/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: getArticleTitle
// ====================================================

export interface getArticleTitle_getArticle {
  __typename: "ArticleDTO";
  title: string | null;
}

export interface getArticleTitle {
  getArticle: getArticleTitle_getArticle | null;
}

export interface getArticleTitleVariables {
  id?: string | null;
  version?: number | null;
}
