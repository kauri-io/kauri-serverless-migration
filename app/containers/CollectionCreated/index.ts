import CollectionCreated from './View.jsx'
import { routeChangeAction } from '../../lib/Epics/RouteChangeEpic'
import { getCollection } from '../../queries/Collection'
import { connect } from 'react-redux'
import { compose, graphql } from 'react-apollo'
import withLoading from '../../lib/with-loading'

const mapStateToProps = () => {
    return {}
}

export default compose(
    connect(
        mapStateToProps,
        { routeChangeAction }
    ),
    graphql(getCollection, {
        options: ({ id }: { id: string }) => ({
            variables: { id },
        }),
    }),
    withLoading()
)(CollectionCreated)
