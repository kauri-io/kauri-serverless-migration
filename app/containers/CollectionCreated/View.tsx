import React from "react";
import styled from "styled-components";
import CollectionCard from "../../components/Card/CollectionCard";
import Link from 'next/link'
import Head from "next/head";
import PrimaryButton from "../../components/Button/PrimaryButton";
import {
  Title2,
  BodyCard,
} from "../../components/Typography";
import {getCollection_getCollection} from '../../queries/__generated__/getCollection'

const Container = styled.section`
  display: flex;
  height: calc(100vh - 76px);
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: ${props => props.theme.colors["textPrimary"]};
  > :first-child {
    margin-bottom: ${props => props.theme.space[1]}px;
    color: white;
  }
  > :nth-child(2) {
    margin-bottom: ${props => props.theme.space[3]}px;
    color: white;
  }
  > :nth-child(3) {
    margin-bottom: ${props => props.theme.space[3]}px;
  }
`;

interface IProps {
  data: {
    getCollection: getCollection_getCollection
  }
  routeChangeAction: (route: string) => void;
  type: string;
}

class CollectionCreated extends React.Component<IProps> {
  render() {
    const {
      data: {
        getCollection: {
          id,
          description,
          dateCreated,
          owner,
          background,
          name,
          sections,
        },
      },
      routeChangeAction,
      type,
    } = this.props;
    const copy = type === "updated" ? "updated" : "live";

    const articleCount =
      sections &&
      sections.reduce((current, next) => {
        if (next && Array.isArray(next.resources)) {
          const articlesInSection = next.resources.filter(
            sectionResource =>
              sectionResource &&
              sectionResource.__typename.toLowerCase().includes("article")
          );
          current += articlesInSection.length;
        }
        return current;
      }, 0);

    const collectionCount =
      sections &&
      sections.reduce((current, next) => {
        if (next && Array.isArray(next.resources)) {
          const collectionsInSection = next.resources.filter(
            sectionResource =>
              sectionResource &&
              sectionResource.__typename.toLowerCase().includes("collection")
          );
          current += collectionsInSection.length;
        }
        return current;
      }, 0);

    const ownerResource =
      owner && owner.__typename === "PublicUserDTO"
        ? {
            avatar: owner.avatar,
            id: owner.id || "not_found",
            type: "USER",
            username: owner.username,
          }
        : owner && owner.__typename === "CommunityDTO"
        ? {
            avatar: owner.avatar,
            id: owner.id || "not_found",
            type: "COMMUNITY",
            username: owner.name,
          }
        : {
            avatar: "",
            id: "",
            username: "",
          };

    return (
      <Container>
        <Head>
          <title>{`Collection ${copy} - Kauri`}</title>
        </Head>
        <Title2>Collection</Title2>
        <BodyCard>{`Your collection is now ${copy}`}</BodyCard>
        <CollectionCard
          id={id}
          description={description}
          date={dateCreated}
          name={name}
          username={ownerResource && ownerResource.username}
          userId={ownerResource && ownerResource.id}
          userAvatar={ownerResource && ownerResource.avatar}
          cardHeight={310}
          imageURL={background}
          articleCount={String(articleCount)}
          collectionCount={String(collectionCount)}
          resourceType={ownerResource.type || "USER"}
          linkComponent={(childrenProps, route) => (
            <Link
              href={route}
            >
              {childrenProps}
            </Link>
          )}
        />
        <PrimaryButton
          onClick={() => routeChangeAction(`/collection/${String(id)}`)}
        >
          View Collection
        </PrimaryButton>
      </Container>
    );
  }
}

export default CollectionCreated;
