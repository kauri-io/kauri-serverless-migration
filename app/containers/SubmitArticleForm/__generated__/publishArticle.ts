/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { ResourceIdentifierInput } from "./../../../../__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: publishArticle
// ====================================================

export interface publishArticle_publishArticle {
  __typename: "MutationResponse";
  hash: string | null;
}

export interface publishArticle {
  publishArticle: publishArticle_publishArticle | null;
}

export interface publishArticleVariables {
  id?: string | null;
  version?: number | null;
  owner?: ResourceIdentifierInput | null;
  signature?: string | null;
  updateComment?: string | null;
}
