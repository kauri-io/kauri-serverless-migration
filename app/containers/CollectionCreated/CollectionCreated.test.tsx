import { shallow } from 'enzyme'
import CollectionCreated from './View'
import { Collection } from '../../mocks'
import { getCollection_getCollection } from '../../queries/__generated__/getCollection'

describe('containers/CollectionCreated/View', () => {
    it('should match the snapshot', () => {
        const props = {
            data: {
                getCollection: Collection.data
                    .getCollection as getCollection_getCollection,
            },
            routeChangeAction: jest.fn(),
            type: 'test',
        }
        const wrapper = shallow(<CollectionCreated {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
