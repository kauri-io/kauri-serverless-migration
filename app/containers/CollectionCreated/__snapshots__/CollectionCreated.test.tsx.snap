// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/CollectionCreated/View should match the snapshot 1`] = `
<View__Container>
  <Head>
    <title>
      Collection live - Kauri
    </title>
  </Head>
  <Typography__styledComponent>
    Collection
  </Typography__styledComponent>
  <Typography__component>
    Your collection is now live
  </Typography__component>
  <CollectionCard
    articleCount="11"
    cardHeight={310}
    collectionCount="2"
    date="2019-04-16T04:39:35.351Z"
    description="Learn about blockchains, smart contracts, Ethereum, and what you can build with them"
    href={
      Object {
        "as": "getting-started/5cb55c871325f2000141df73/c",
        "href": "/collection?collection_id=5cb55c871325f2000141df73",
      }
    }
    id="5cb55c871325f2000141df73"
    imageURL="https://api.kauri.io:443/ipfs/QmQTsd7FhCLBQVJCYNP8a5Vf1G68AsxLPyDtWP3gToc7g4"
    name="Getting Started"
    resourceType="USER"
    userAvatar="https://api.beta.kauri.io:443/ipfs/QmRVCyQ3ng5AWGmjodzXcxg1LK9CRvePmd6ciDXY1mLofY"
    userId="37648fc15a8365735289e002d65d44d80c505e8b"
  />
  <PrimaryButtonComponent
    onClick={[Function]}
  >
    View Collection
  </PrimaryButtonComponent>
</View__Container>
`;
